.github/workflows/artifact-tests.yml

```diff
@@ -0,0 +1,144 @@
+    - name: Upload artifacts using uploadArtifact()
+      run: |
+        export ACTIONS_RUNTIME_TOKEN="${{ steps.get_runtime_token.outputs.EnvValue }}"
+        export ACTIONS_RUNTIME_URL="${{ steps.get_runtime_url.outputs.EnvValue }}"
+        export GITHUB_RUN_ID="${{ steps.get_run_id.outputs.EnvValue }}"
+        node -e "Promise.resolve(require('./packages/artifact/lib/artifact-client').create().uploadArtifact('my-artifact-1',['artifact-path/world.txt'], '${{ github.workspace }}'))"
+        node -e "Promise.resolve(require('./packages/artifact/lib/artifact-client').create().uploadArtifact('my-artifact-2',['artifact-path/gzip.txt'], '${{ github.workspace }}'))"
+
+    - name: Download artifacts using downloadArtifact()
+      run: |
+        export ACTIONS_RUNTIME_TOKEN="${{ steps.get_runtime_token.outputs.EnvValue }}"
+        export ACTIONS_RUNTIME_URL="${{ steps.get_runtime_url.outputs.EnvValue }}"
+        export GITHUB_RUN_ID="${{ steps.get_run_id.outputs.EnvValue }}"
+        mkdir artifact-1-directory
+        node -e "Promise.resolve(require('./packages/artifact/lib/artifact-client').create().downloadArtifact('my-artifact-1','artifact-1-directory'))"
+        mkdir artifact-2-directory
+        node -e "Promise.resolve(require('./packages/artifact/lib/artifact-client').create().downloadArtifact('my-artifact-2','artifact-2-directory'))"
+    
+    - name: Verify downloadArtifact()
+      run: |
+        $file1 = "artifact-1-directory/artifact-path/world.txt"
+        if(!(Test-Path -path $file1))
+        {
+            Write-Error "artifact-1-directory/artifact-path/world.txt file does not exist"
+        }
+        if(!((Get-Content $file1) -ceq 'hello'))
+        {
+            cat $file1
+            Write-Error "File contents of world.txt are incorrect"
+        }
+
+        $file2 = "artifact-2-directory/artifact-path/gzip.txt"
+        if(!(Test-Path -path $file2))
+        {
+            Write-Error "artifact-2-directory/artifact-path/gzip.txt does not exist"
+        }
+        if(!((Get-Content $file2) -ceq 'Some large amount of text that has a compression ratio that is greater than 100%. If greater than 100%, gzip is used to upload the file'))
+        {
+            cat $file2
+            Write-Error "File contents of gzip.txt are incorrect"
+        }
+      shell: pwsh
+
+    - name: Download artifacts using downloadAllArtifacts()
+      run: |
+        export ACTIONS_RUNTIME_TOKEN="${{ steps.get_runtime_token.outputs.EnvValue }}"
+        export ACTIONS_RUNTIME_URL="${{ steps.get_runtime_url.outputs.EnvValue }}"
+        export GITHUB_RUN_ID="${{ steps.get_run_id.outputs.EnvValue }}"
+        mkdir multi-artifact-directory
+        node -e "Promise.resolve(require('./packages/artifact/lib/artifact-client').create().downloadAllArtifacts('multi-artifact-directory'))"
+
+    - name: Verify downloadAllArtifacts() 
+      run: |
+        $file1 = "multi-artifact-directory/my-artifact-1/artifact-path/world.txt"
```

Comment:
We should include these as separate script files in the repo rather than having the entire thing in the workflow file

Reply:
What benefit does putting the scripts in separate files have here? Is the entire script in a single workflow always a bad thing?

If the scripts start to get really really long (or if they need to be reused by other workflows) then I can see it being desired, but in this scenario it would just make it hard to maintain without a real benefit. Having worked with some of our larger canaries that used a lot of templates and script references, it gets really annoying to have to dig through different files. Here if a run fails, you can just look at the `WorkFlow file` tab and you have all your code.

Also, think about the artifact names, contents and paths that we are checking here (what we specify in other steps). We would have to cross-reference the file paths, contents and the artifact names in the script files and the YAML. Considering the simplicity of what we're trying to do, I absolutely think that separate script files are not warranted.

Reply:
The large script blocks make it hard to parse the workflow, and you're duplicating a lot of the checking logic.

For example, looking at checkout's E2E tests, it's very easy to read the workflow and understand what's being validated: https://github.com/actions/checkout/blob/master/.github/workflows/test.yml#L175

The scripts themselves are very short (https://github.com/actions/checkout/blob/master/__test__/verify-basic.sh) but the actual script content isn't necessary to understand what's happening in the workflow.

>We would have to cross-reference the file paths, contents and the artifact names in the script files and the YAML. 

These could all be passed in as environment variables or command-line arguments instead of being hardcoded in the scripts

Reply:
Took a wee bit of work, but it's done